marshal.go

```diff
@@ -636,26 +644,30 @@ func (intTypeInfo) Marshal(value interface{}) ([]byte, error) {
 
 // Unmarshal unmarshals the byte slice into the value.
 func (i intTypeInfo) Unmarshal(data []byte, value interface{}) error {
+	decodedData, err := decInt(data)
+	if err != nil {
+		return unmarshalErrorf("%s", err.Error())
+	}
 	if iptr, ok := value.(*interface{}); ok && iptr != nil {
 		var v int
-		if err := unmarshalIntlike(TypeInt, int64(decInt(data)), data, &v); err != nil {
+		if err := unmarshalIntlike(TypeInt, int64(decodedData), data, &v); err != nil {
 			return err
 		}
 		*iptr = v
 		return nil
 	}
-	return unmarshalIntlike(TypeInt, int64(decInt(data)), data, value)
+	return unmarshalIntlike(TypeInt, int64(decodedData), data, value)
 }
 
 func encInt(x int32) []byte {
 	return []byte{byte(x >> 24), byte(x >> 16), byte(x >> 8), byte(x)}
 }
 
-func decInt(x []byte) int32 {
+func decInt(x []byte) (int32, error) {
 	if len(x) != 4 {
-		return 0
+		return 0, fmt.Errorf("expected 4 bytes decoding int but got %v", len(x))
```

Comment:
I'm confused, isn't this going to make `NULL` now return an error instead of `0` like it did before?

Reply:
> I'm confused, isn't this going to make NULL now return an error instead of 0 like it did before?

Yeah I started seeing a couple of test failures and I noticed that, was just working on it now. Should be fixed now.